packages/venia-concept/src/components/CreateAccount/createAccount.js

```diff
@@ -0,0 +1,220 @@
+                        <span>An account gives you access to rewards!</span>
+                    </div>
+
+                    <Input
+                        onChange={this.updateEmail}
+                        selected={true}
+                        label={'Email'}
+                        helpText={emailHelpText}
+                        helpType={emailHelpType}
+                        required={true}
+                    />
+
+                    <Input
+                        onChange={this.updateFirstName}
+                        label={'First Name'}
+                        required={true}
+                    />
+
+                    <Input
+                        onChange={this.updateLastName}
+                        label={'Last Name'}
+                        required={true}
+                    />
+
+                    <Input
+                        onChange={this.updatePassword}
+                        label={'Password'}
+                        type={'password'}
+                        required={true}
+                        placeholder={'Enter a password'}
+                        helpText={
+                            'Password must be at least 8 characters long and contain 3 or more of the following: Lowercase, Uppercase, Digits, or Special Characters. (ex. Password1)'
+                        }
+                    />
+
+                    <Input
+                        onChange={this.updatePasswordConfirm}
+                        label={'Confirm Password'}
+                        type={'password'}
+                        required={true}
+                        placeholder={'Enter the password again'}
+                        helpText={passwordHelpText}
+                        helpType={passwordHelpType}
+                    />
+
+                    <Checkbox
+                        label={'Subscribe to news and updates'}
+                        select={this.handleCheckboxChange}
+                        initialState={this.state.subscribe}
+                    />
+                    <div className={classes.createAccountButton}>
+                        <Button type="submit" disabled={disableAccountCreation}>
+                            Create Account
+                        </Button>
+                    </div>
+                    {errorMessage}
+                </Form>
+            </div>
+        );
+    }
+
+    onCreateAccount = () => {
+        if (!this.disableAccountCreation) {
+            const newCustomer = {
+                customer: {
+                    firstname: this.state.firstName,
+                    lastname: this.state.lastName,
+                    email: this.state.email
+                },
+                password: this.state.password
+            };
+            this.props.createAccount(newCustomer);
+        }
+    };
+
+    handleCheckboxChange = value => {
+        this.setState({ subscribe: value });
+    };
+
+    checkEmail = debounce(async email => {
```

Comment:
Do you want this debounced on the prototype, or debounced per instance? Not that there will be multiple instances active at once in the near future, but if there are, the debounce will only exist per-instance.

Reply:
I'm ok leaving the debounce on the prototype - this way we are ensuring that the requests be limited by the debounce, instead of sending a new request every key press. Is that OK? I'm open to changing it.

Reply:
In your code, the debounce is on the instance. The class assignment shorthand means that this method is created per-instance.

It's totally fine--just wanted to make sure it was intentional. Usually the mistake goes in the other direction; a debounce on the prototype causes an arguably worse problem where individual calls from individual instances are lost. I put together a [Codepen to demonstrate.](https://codepen.io/zetlen/pen/zmGbyq?editors=0012)